newshomepages/save_html.py

```diff
@@ -0,0 +1,130 @@
+    height: str = "1600",
+    full_page: bool = False,
+):
+    """Save HTML for the provided homepage."""
+    site = utils.get_site(handle)
+    output_path = Path(output_dir)
+    _save_html(
+        site,
+        output_path,
+        wait=int(wait),
+        width=int(width),
+        height=int(height),
+        full_page=bool(full_page),
+    )
+
+
+@retry(tries=3, delay=5, backoff=2)
+def _save_html(
+    site: typing.Dict,
+    output_path: Path,
+    wait: int = 5000,
+    width: int = 1300,
+    height: int = 1600,
+    full_page: bool = False,
+):
+    """Shoot the provided site."""
+    print(f":camera: Fetching HML for {site['name']}")
+
+    # Set the output path
+    output_path.mkdir(parents=True, exist_ok=True)
+
+    with sync_playwright() as playwright:
+        context = utils._load_persistent_context(playwright, width, height)
+        page = utils._load_new_page_disable_javascript(
+            context=context,
+            url=site["url"],
+            wait_seconds=int((site["wait"] or wait) / 1000),
+            handle=site["handle"],
+            full_page=full_page,
+        )
+
+        single_file_pre_load_extensions = [
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-bootstrap.js"
+            ),
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-hooks-frames.js"
+            ),
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-frames.js"
+            ),
+        ]
+        for f in single_file_pre_load_extensions:
+            utils.read_script_from_file(f, page)
+
+        post_load_script = str(
+            utils.EXTENSIONS_PATH / "singlefile" / "javascript" / "single-file.js"
+        )
+        utils.read_script_from_file(post_load_script, page)
+        page_html_content = page.evaluate(
+            """
+                       () => singlefile.getPageData({
+                         removeHiddenElements: true,
+                         removeUnusedStyles: true,
+                         removeUnusedFonts: true,
+                         removeImports: true,
+                         blockScripts: true,
+                         blockAudios: true,
+                         blockVideos: true,
+                         compressHTML: true,
+                         removeAlternativeFonts: true,
+                         removeAlternativeMedias: true,
+                         removeAlternativeImages: true,
+                         groupDuplicateImages: true
+                       });
+                   """
+        ).get("content")
+
+        if page_html_content is None:
+            print("NO HTML CONTENT!")
+
+        html_file_path = str(output_path / f"{site['handle'].lower()}.html")
```

Comment:
Does this need to converted with `str`?

Reply:
yeah this was a weird one. it was being interpreted as a very different variable. I'll check again. Was late at night so maybe I was seeing things...